src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -185,8 +185,14 @@ def _upgrade_on_windows():
     Directly installing from URL may be blocked by policy: https://github.com/Azure/azure-cli/issues/19171
     This also gives the user a chance to manually install the MSI in case of msiexec.exe failure.
     """
-    logger.warning("Updating Azure CLI with MSI from https://aka.ms/installazurecliwindows")
-    tmp_dir, msi_path = _download_from_url('https://aka.ms/installazurecliwindows')
+    import platform
+
+    if platform.architecture()[0] == '32bit':
```

Comment:
[`platform.architecture`](https://docs.python.org/3/library/platform.html#platform.architecture) returns the bit architecture of the Python executable:

```py
> C:\Users\xxx\AppData\Local\Programs\Python\Python310-32\python.exe -c "import platform; print(platform.architecture())"
('32bit', 'WindowsPE')
> C:\Users\xxx\AppData\Local\Programs\Python\Python310\python.exe -c "import platform; print(platform.architecture())"
('64bit', 'WindowsPE')
```

Do we want to detect the machine's bit architecture instead?

Reply:
[`platform.machine`](https://docs.python.org/3/library/platform.html#platform.machine) seems to be the candidate for detecting the machine's architecture:

```
> & $env:LOCALAPPDATA\Programs\Python\Python310-32\python.exe -c "import platform; print(platform.machine())"
AMD64
> & $env:LOCALAPPDATA\Programs\Python\Python310\python.exe -c "import platform; print(platform.machine())"
AMD64
```

Reply:
No, I guess most users are using a 64-bit OS. I don't want to force them use 64-bit and break their extensions unexpectedly.

Reply:
Fair enough.